### ISSUE TITLE
Cleanup OCP\JSON and OC_JSON usage

### ISSUE THREAD

**Issue by rullzer** [nextcloud/server#8568]
Since this is deprecated API we should really get rid of it sooner rather than later! Especially since it returns invalid status codes for errors (200 usually).

For checkedLoggedIn:

- files_version:
  - [ ] https://github.com/nextcloud/server/blob/0215b004da6af887b3e3efa28c425d41a1c76b9b/apps/files_trashbin/ajax/delete.php#L27
  - [ ] https://github.com/nextcloud/server/blob/70b1f510f2eab2276beab2e0b47d08f04c63c966/apps/files_versions/download.php#L29
  - [ ] https://github.com/nextcloud/server/blob/813f0a0f40dd42b28cd35d91616f3f87ef400861/apps/files_versions/ajax/getVersions.php#L27
  - [ ] https://github.com/nextcloud/server/blob/813f0a0f40dd42b28cd35d91616f3f87ef400861/apps/files_trashbin/ajax/list.php#L24
  - [ ] https://github.com/nextcloud/server/blob/0215b004da6af887b3e3efa28c425d41a1c76b9b/apps/files_versions/ajax/rollbackVersion.php#L29
- files_trashbin:
  - [ ] https://github.com/nextcloud/server/blob/813f0a0f40dd42b28cd35d91616f3f87ef400861/apps/files_trashbin/ajax/isEmpty.php#L25
  - [ ] https://github.com/nextcloud/server/blob/0eebff152a177dd59ed8773df26f1679f8a88e90/apps/files_trashbin/ajax/undelete.php#L29
- files_external:
  - [ ] https://github.com/nextcloud/server/blob/0eebff152a177dd59ed8773df26f1679f8a88e90/apps/files_external/ajax/oauth2.php#L26
- files:
  - [ ] https://github.com/nextcloud/server/blob/0215b004da6af887b3e3efa28c425d41a1c76b9b/apps/files/ajax/getstoragestats.php#L32
  - [ ] https://github.com/nextcloud/server/blob/eb51f06a3b9e42686f462b9f7a56411d3fe6cb27/apps/files/ajax/list.php#L30
- core
  - [ ] https://github.com/nextcloud/server/blob/0215b004da6af887b3e3efa28c425d41a1c76b9b/core/search/ajax/search.php#L29

There is probably more I missed but this is a good start to cleanup.

**Comment by MorrisJobke** [comment]
@schiessle as we talked about trashbin and versions ;)

**Comment by juliushaertl** [comment]
- [ ] settings/ajax/* uses a lot of OCP\JSON and OC_JSON as well
  - for apps management this will be addressed in https://github.com/nextcloud/server/issues/7931

**Comment by rullzer** [comment]
@MorrisJobke yes versions and trashbin should be cleaned up. But IMO they should get an overhaul anyway,

* Move to a dav endpoint (so web+clients can all use it)
* Move away from all the static code maddness

**Comment by szaimen** [comment]
I suppose this issue is still valid? If not, please close this issue!

**Comment by juliushaertl** [comment]
Still valid

**Comment by q-wertz** [comment]
I think also `user_ldap` still relies on this. See e.g. https://github.com/nextcloud/server/issues/27871 (resp. comment https://github.com/nextcloud/server/issues/27871#issuecomment-895846733)
